flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/functions/aggfunctions/MaxWithRetractAggFunction.java

```diff
@@ -135,6 +135,8 @@ private void updateMax(MaxWithRetractAccumulator<T> acc) throws Exception {
 		// when both of them are expired.
```

Comment:
I am confused about this comments, expired?
If it is an empty map, why there is a max?
Should set max to null at the moment of becoming empty?

Reply:
Yes. This is the problem. We should set max to null when the map is empty.

The "expired" means the content of MapView might be cleared because of state ttl.